File: game-server/src/main/java/org/forfun/mmorpg/game/ServerType.java
package org.forfun.mmorpg.game;

public enum ServerType {

    GATE(0, "网关服"),

    GAME(1, "游戏服"),

    FIGHT(2, "战斗服"),

    ;

    ServerType(int type, String name) {
        this.type = type;
        this.name = name;
    }

    int type;

    String name;

    public static ServerType of(int type) {
        for (ServerType value : ServerType.values()) {
            if (value.type == type) {
                return value;
            }
        }
        return null;
    }
}

File: game-server/src/main/java/org/forfun/mmorpg/game/ServerType.kt
package org.forfun.mmorpg.game

enum class ServerType(val type: Int, val desc: String) {

    GATE(0, "网关服"),

    GAME(1, "游戏服"),

    FIGHT(2, "战斗服"),

    CENTRE(10, "中心服");

    companion object {
        fun of(type: Int): ServerType? {
            return entries.find { it.type == type }
        }
    }
}
